### ISSUE TITLE
[BUG]: PostgreSQL ORDER BY and NULLS FIRST, NULLS LAST

### ISSUE THREAD

**Issue by tztztztz** [phalcon/cphalcon#15917]
**Describe the bug**
I'm not sure if it's bug or feature request but I'm leaning more towards bug, so I'm marking it as a bug.

In PostgreSQL you can decide how to treat NULLS when ordering records: NULLS FIRST, NULLS LAST.

Examples:

SELECT a.* FROM my_table a ORDER BY a.total_sum DESC NULLS LAST;
SELECT * FROM current_store ORDER BY qty ASC NULLS FIRST;

PostgreSQL documentation:

https://www.postgresql.org/docs/current/queries-order.html

When trying to use it with Phalcon ORM (PostgreSQL dialect), it produces error:

**Syntax error, unexpected token IDENTIFIER(NULLS), near to ' LAST LIMIT :APL0: OFFSET :APL1:'**

**To Reproduce**
Just use Phalcon ORM builder and try to order by [ASC|DESC] [NULLS LAST|NULLS FIRST]

```php
$builder->orderBy('DESC NULLS LAST);
$builder->orderBy('DESC NULLS FIRST);
$builder->orderBy('ASC NULLS LAST);
$builder->orderBy('ASC NULLS FIRST);
```

**Comment by BeMySlaveDarlin** [comment]
Right now pgsql part of phalcon orm quite poor and not all features implemented.
But. It has Criteria feature, which can be used to implement custom expressions.

**Comment by tztztztz** [comment]
Yes, and I previously contributed to Phalcon ORM and fixed minor issue regarding PostgreSQL, maybe I will fix it too, but I'm not too good with all the building process and github at all, but if it not gonna be fixed in like 2-3 months I'll probably will try.

**Comment by BeMySlaveDarlin** [comment]
RN we team is out of resourses for extending orm and implementing new features due to finishing v5 asap.
It will be gr8, if you create non-bc pull request with such a fix.

**Comment by tztztztz** [comment]
When I'll find the Time I will try, right now I fixed my problem by other way around.

**Comment by hjdfdr44** [comment]
I even have any blunders for the duration of compilation. Second in my enterprise I would like install handiest more strong Phalcon like four.1.2 which I applied on [celebspanda](http://celebspanda.com/). The best choice for me could be if you release new version of Phalcon 4 with this computer virus.

**Comment by niden** [comment]
I am marking this won't fix. There is no support for this syntax and it is only postgres specific so it will not work with other adapters.

Thank you for the report.

**Comment by tztztztz** [comment]
The NULLS FIRST | NULLS LAST syntax is actually part of the SQL standard not a specific PostgreSQL feature

https://docs.datavirtuality.com/pipes/order-by-clause

It is since 2003, and was added in ISO/IEC 9075:2003 specification, and it is part of the standard today.

That feature is sometimes very useful.

Please unblock and mark as bug.

**Comment by niden** [comment]
I do not know about the standard you mention, I go by with what i supported and unlike Postgres or Oracle, MySQL does not support this syntax. One has to use the `IS NULL` to achieve the same result.

This is not a bug. It is a feature that Postgres has that we do not support. You are more than welcome to open a new New Feature Request issue, and if the community wants it, we will implement it.
